### PR TITLE
Instantiate job store

### PR DESCRIPTION
**Pull request by ewjoachim**
Based on #49

This PR:
- Finally makes up cabbages's mind regarding App instanciation: it takes an instance of JobStore
- Adds __version__ and other metadata at the top level
- Add building of the docs in the CI, we realized it was missing

### CODE REVIEW

File: cabbage/retry.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    Parameters
+    ----------
+    max_attempts:
+        The maximum number of attempts the job should be retried
+    wait:
+        The number of seconds to wait between attempts
+    """
+
+    max_attempts: Optional[int] = None
+    wait: int = 0
```
Comment (mgu) [inline cabbage/retry.py]: is it a safe default ?

Comment (ewjoachim) [reply cabbage/retry.py]: If you just ask to "retry" without specifying to wait and without specifying how many times, I'd say yes. Who know if 1s is more appropriate than 1 day. Who knows if 3 times is more appropriate than 300 times ?

Comment (ewjoachim) [reply cabbage/retry.py]: Worse case scenario, we'll... Try. again and again and again.

Comment (ewjoachim) [reply cabbage/retry.py]: Tenacity does exactly the same, and they basically know about retry decorators ! https://tenacity.readthedocs.io/en/latest/#basic-retry :)

**Pull request merged**